lib/jsdom/living/nodes/GlobalEventHandlers-impl.js

```diff
@@ -1,6 +1,202 @@
+
+const idlUtils = require("../generated/utils");
+
+const ErrorEvent = require("../generated/ErrorEvent");
+
+const events = new Set([
+  "abort", "autocomplete",
+  "autocompleteerror", "blur",
+  "cancel", "canplay", "canplaythrough",
+  "change", "click",
+  "close", "contextmenu",
+  "cuechange", "dblclick",
+  "drag", "dragend",
+  "dragenter", "dragexit",
+  "dragleave", "dragover",
+  "dragstart", "drop",
+  "durationchange", "emptied",
+  "ended", "error", "focus",
+  "input", "invalid",
+  "keydown", "keypress",
+  "keyup", "load", "loadeddata",
+  "loadedmetadata", "loadstart",
+  "mousedown", "mouseenter",
+  "mouseleave", "mousemove",
+  "mouseout", "mouseover",
+  "mouseup", "wheel",
+  "pause", "play",
+  "playing", "progress",
+  "ratechange", "reset",
+  "resize", "scroll",
+  "seeked", "seeking",
+  "select", "show",
+  "sort", "stalled",
+  "submit", "suspend",
+  "timeupdate", "toggle",
+  "volumechange", "waiting"
+]);
+
+const inheritedAttributes = new Set(["blur", "error", "focus", "load", "resize", "scroll"]);
+
+function appendHandler(el, eventName) {
+  el.addEventListener(eventName, event => {
+    event = idlUtils.implForWrapper(event);
+
+    const callback = el["on" + eventName];
+    if (!callback) {
+      return;
+    }
+
+    let returnValue = null;
+    if (ErrorEvent.isImpl(event)/* && callback is OnErrorEventHandler*/) {
+
+    } else {
+      returnValue = callback.call(event.currentTarget, event);
+    }
+
+    if (eventName === "mouseover" || (eventName === "error" && ErrorEvent.isImpl(event))) {
+      if (returnValue) {
+        event._canceledFlag = true;
+      }
+    // TODO: before unload
+    } else if (!returnValue) {
+      event._canceledFlag = true;
+    }
+  }, false);
+}
+
+function createEventMethod(obj, event) {
+  Object.defineProperty(obj, "on" + event, {
+    get() {
+      if (inheritedAttributes.has(event) && (this.tagName === "BODY" || this.tagName === "FRAMESET")) {
+        return this.ownerDocument.defaultView["on" + event] === undefined ? null : this.ownerDocument.defaultView["on" + event];
+      }
+
+      const value = this._eventHandlers[event];
+      if (!value) {
+        return null;
+      }
+
+      if (value.body !== undefined) {
+        let element;
+        let formOwner = null;
+        let document;
+        let window;
+        if (this.constructor.name === "Window") {
+          element = null;
+          window = this;
+          document = idlUtils.implForWrapper(this.document);
+        } else {
+          element = this;
+          document = element.ownerDocument;
+          window = document.defaultView;
+        }
+        const body = value.body;
+        // TODO: location
+        if (element !== null) {
+          formOwner = element.form || null;
+        }
+
+        try {
+          // eslint-disable-next-line no-new-func
+          Function(body); // properly error out on syntax errors
+        } catch (e) {
+          // TODO: Report the error
```

Comment:
Shouldn't be that hard with the utils we have; I think it's in the deleted code.